### PR TITLE
Fixing FW issues when pattern is longer than 15 characters on g++ > 5.2

### PR DESCRIPTION
**Pull request by jwlodek**
Please see https://epics.anl.gov/tech-talk/2020/msg00515.php for detailed description of issue

### CODE REVIEW

**Comment by jwlodek** [comment]
Does it need the `-1`? In the c++ reference they just do `sizeof()`:

```
char str1[]= "To be or not to be";
char str2[40];
char str3[40];

 /* copy to sized buffer (overflow safe): */
 strncpy ( str2, str1, sizeof(str2) );
```

**Comment by MarkRivers** [comment]
This is from "man snprintf".
```
The strncpy() function is similar, except that at most n bytes of src are copied.  Warning: If there is no null byte among the first n bytes of src, the string  placed
       in dest will not be null-terminated.
```
This means you should could omit the -1 but you need to set acq.pattern[MAX_BUF_LEN-1]=0 after the strncpy to make sure the string is null-terminated.

**Comment by jwlodek** [comment]
Yes I just noticed that on the reference page. I added the -1 for simplicity.

**Comment by MarkRivers** [comment]
The -1 does not guarantee null-termination unless the buffer was originally set to all 0.

File: eigerApp/src/eigerDetector.cpp [hunk 1]
```diff
@@ -740,7 +740,9 @@ void eigerDetector::controlTask (void)
         if(dataSource == SOURCE_FILEWRITER || (fwEnable && saveFiles))
         {
             acquisition_t acq;
```
Comment (brunoseivam) [inline eigerApp/src/eigerDetector.cpp]: I believe changing this line to `acquisition_t acq = {};` will take care of zero-inizializing `acq` and all its fields.

**Comment by jwlodek** [comment]
Updated to guarantee null terminating string after copy

**Pull request merged**